docker-compose.yml

```diff
@@ -1,20 +1,27 @@
----
+version: '3.8'
 services:
-  postgres:
+  db:
     image: postgres:14
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
     environment:
-      POSTGRES_USER: ${DATABASE_USERNAME}
-      POSTGRES_PASSWORD: ${DATABASE_PASSWORD}
-      POSTGRES_DB: pet_rescue_development
-
-    ports:
-      - 5432
+      POSTGRES_DB: myapp_development
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: password
```

Comment:
it's good to go ahead and set a default value in here, but we still will want to allow overriding these values by setting environment variables.

There's a not-uncommon pattern of running support services, like postgres and redis, via a dockerfile, and even running the local dev server within docker, but still running tests in the non-docker context.  Respecting set envvars keeps friction lower.

```
POSTGRES_USER: ${DATABASE_USERNAME:-postgres}
```
ought to work

Reply:
Ah I like this.